### PR TITLE
transport/header: tests

### PR DESCRIPTION
**Pull request by abhinav**
Depends on #62.

[Review on Reviewable](https://reviewable.io/reviews/yarpc/yarpc-go/66)

### CODE REVIEW

**Comment by breerly** [comment]
:shipit:

**Comment by prashantv** [comment]
lgtm

**Comment by abhinav** [comment]
Rebased. Blank values are now allowed.

File: transport/header_test.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewHeaders(t *testing.T) {
+	tests := []struct {
+		headers  map[string]string
+		matches  map[string]string
+		failures map[string]struct{}
```
Comment (prashantv) [inline transport/header_test.go]: I think you actually want `[]string` here since you just iterate over failures and make sure they're not in the map

**Comment by prashantv** [comment]
lgtm with minor comment

File: transport/header_test.go (hunk 1 above)
Comment (abhinav) [inline transport/header_test.go]: :+1:

**Pull request merged**